### PR TITLE
Added ability to filter docs for specific runner

### PR DESCRIPTION
**Pull request by doesitblend**
### What does this PR do?
Updates `doc.runner` to make use of the already underlying functionality to supply an argument to look at docs for a specific runner.

### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Calling `salt-run doc.runner` would dump all runner docs

### New Behavior
You can now call `salt-run doc.runner winrepo` or `salt-run doc.runner manage.down` to get docs specific to that module, or method.

### CODE REVIEW

**Comment by max-arnold** [comment]
Maybe it also makes sense to do the same for `doc.wheel` and `doc.execution` functions for consistency purposes?

**Comment by waynew** [comment]
Hey @doesitblend I went ahead and wrote some tests here: https://github.com/waynew/salt/pull/new/tc/update-doc-runner-57757

File: tests/pytests/unit/runners/test_doc.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import pytest
+import salt.runner
+import salt.runners.doc as doc
+from tests.support.mock import PropertyMock, patch
+
+
+@pytest.fixture(autouse=True)
+def setup_loader(request):
```
Comment (s0undt3ch) [inline tests/pytests/unit/runners/test_doc.py:8]: This needs to be updated to the new way of setting loader modules.
Please check the other tests on how.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**